src/metabase/models/values_card.clj

```diff
@@ -0,0 +1,83 @@
+(ns metabase.models.values-card
+  (:require
+   [clojure.string :as str]
+   [metabase.query-processor :as qp]
+   [metabase.util :as u]
+   [toucan.db :as db]
+   [toucan.models :as models]))
+
+
+(models/defmodel ValuesCard :values_card)
+
+(u/strict-extend #_{:clj-kondo/ignore [:metabase/disallow-class-or-type-on-model]} (class ValuesCard)
+                 models/IModel
+                 (merge models/IModelDefaults
+                        {:properties (constantly {:timestamped? true})
+                         :types      (constantly {:parameterized_object_type :keyword})}))
+
+(defn- search-match?
+  [search-term filter-term]
+  (str/includes? filter-term search-term))
+
+(defn query-matches
+  "Filter the values according to the `search-term`. If `search-term` is blank, return all the values without filtering"
+  [search-term values]
+  (if (str/blank? search-term)
+    values
+    (let [normalize              (comp str/lower-case str/trim)
+          normalized-search-term (normalize search-term)]
+      (filter #(search-match? normalized-search-term (normalize %)) values))))
+
+(defn- query-for-dashboard
+  [{dashboard-id :id} param-key]
+  (->> (db/query {:select [:dataset_query]
+                  :from [:report_card]
+                  :join [:values_card [:= :report_card.id :values_card.card_id]]
+                  :where [:and
+                          [:= :parameterized_object_id dashboard-id]
+                          [:= :parameterized_object_type "dashboard"]
+                          [:= :parameter_id param-key]]})
+       (db/do-post-select 'Card)
+       first
+       :dataset_query))
+
+(defn values-for-dashboard
+  "Returns a map with the filter `:values` associated with the dashboard."
+  [dashboard param-key search-term]
+  {:values
```

Comment:
this should have a key `has_more_values` we could default it to `false` for now.

But we should set it to true if the number of values returned = query.limit. This flag is used by UI to determine when users type search in the parameter widget, whether or not they need to call to BE to search, or they can just do in-browser search.

Reply:
good call, added